conftest.py

```diff
@@ -388,74 +415,35 @@ def old_bare_repository(request, tmpdir_factory):
 )
 def old_workflow_project(request, tmp_path_factory):
     """Prepares a testing repo created by old version of renku."""
-    import tarfile
-    from pathlib import Path
-
-    from git import Repo
-
     name = request.param["name"]
-
-    compressed_repo_path = Path(__file__).parent / "tests" / "fixtures" / "{name}.tar.gz".format(name=name)
-
-    working_dir_path = tmp_path_factory.mktemp(name)
-
-    with tarfile.open(str(compressed_repo_path), "r") as fixture:
-        fixture.extractall(str(working_dir_path))
-
-    path = working_dir_path / name
-
-    repo_path = tmp_path_factory.mktemp("repo")
-
-    repository = Repo(str(path), search_parent_directories=True).clone(str(repo_path))
-
+    base_path = tmp_path_factory.mktemp(name)
+    repository = clone_compressed_repository(base_path=base_path, name=name)
     repository_path = repository.working_dir
 
-    commit = repository.head.commit
-
     os.chdir(repository_path)
+
     yield {
         "repo": repository,
         "path": repository_path,
         "log_path": request.param["log_path"],
         "expected_strings": request.param["expected_strings"],
     }
-    os.chdir(repository_path)
-    repository.head.reset(commit, index=True, working_tree=True)
-    # remove any extra non-tracked files (.pyc, etc)
-    repository.git.clean("-xdff")
-
-    shutil.rmtree(str(repo_path))
-
-
-@pytest.fixture(scope="module")
-def old_repository(tmpdir_factory, old_bare_repository):
-    """Create git repo of old repository fixture."""
-    import shutil
-
-    from git import Repo
 
-    repo_path = tmpdir_factory.mktemp("repo")
-    yield {
-        "repo": Repo(old_bare_repository["path"].strpath, search_parent_directories=True).clone(repo_path.strpath),
-        "exit_code": old_bare_repository["exit_code"],
-    }
-    shutil.rmtree(repo_path.strpath)
+    shutil.rmtree(base_path)
 
 
 @pytest.fixture
-def old_project(old_repository):
-    """Create a test project."""
-    repo = old_repository["repo"]
-    repository_path = repo.working_dir
+def old_dataset_project(tmp_path_factory):
+    """Prepares a testing repo created by old version of renku."""
+    name = "old-datasets-v0.9.1.git"
```

Comment:
I was thinking about this before. Should we make this a parameter?

Reply:
This one is a bit different than the others because we do more checks on this repository after migration. This fixture is function-scoped (because it is used in only one test) while the others are session-scoped. I am not sure if we can parameterize this along with the others in a clean way. Do you have some solutions in mind?

Reply:
I see. I guess it's fine then. I believe this might get refactored in the near future though since we might add more old projects created with different versions. The way I see it that we might refactor it as a parametrized test is that alongside testing fixture we also provide expected snapshots of metadata for that fixture. 

Then we can write a test which would:
1) ensure that fixture metadata and provided snapshot are actually different
2) run migration
3) ensure that fixture metadata and provided snapshot are the same

Usually, this kind of snapshot testing is done in the UI, but I think if it works fine for HTML it will work fine for our YAML files as well. But anyway, this is way out of scope here, so we can create a ticket for this if you agree.